e2e/test/ASCforIoTOmsClient.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    public class ASCforIoTOmsClient : IDisposable
+    {
+        private const string AuthenticationAuthorityTemplate = "https://login.windows.net/{0}";
+        private const string Audience =  "https://api.loganalytics.io/";
+        private const string QueryUriTemplate = "https://api.loganalytics.io/{0}/workspaces/{1}/query";
+        private const string LogAnalyticsApiVersion = "v1";
+        private const string RawEventQueryTemplate =
+            @"SecurityIoTRawEvent
+    | where DeviceId == ""{0}""
+    | where IoTRawEventId == ""{1}""";
+        
+        private readonly string _workspaceId = Configuration.Oms.WorkspacedId;
+        private readonly string _aadTenant = Configuration.Oms.AadTenant;
+        private readonly string _appId = Configuration.Oms.AadAppId;
+        private readonly string _appKey = Configuration.Oms.AadAppKey;
+
+        private readonly HttpClient _client;
+        private readonly AuthenticationContext _authenticationContext;
+        private readonly string _queryUri;
+
+        public static ASCforIoTOmsClient CreateClient()
+        {
+            return new ASCforIoTOmsClient();
+        }
+
+        private ASCforIoTOmsClient()
+        {
+            _client = new HttpClient();
+            string authority = string.Format(CultureInfo.InvariantCulture, AuthenticationAuthorityTemplate, _aadTenant);
+            _authenticationContext = new AuthenticationContext(authority);
+            _queryUri = string.Format(CultureInfo.InvariantCulture, QueryUriTemplate, LogAnalyticsApiVersion, _workspaceId);
+        }
+
+        public async Task<bool> IsRawEventExist(string deviceId, string eventId)
+        {
+            bool isEventExist = false;
+            string query = string.Format(CultureInfo.InvariantCulture, RawEventQueryTemplate, deviceId, eventId);
+            var sw = new Stopwatch();
+            sw.Start();
+            while (!isEventExist && sw.Elapsed.TotalMinutes < 30)
+            {
+                isEventExist = await DoQuery(query).ConfigureAwait(false);
+                await Task.Delay(TimeSpan.FromSeconds(20)).ConfigureAwait(false);
+            }
+
+            sw.Stop();
+            return isEventExist;
+        }
+
+        private async Task<bool> DoQuery(string query)
+        {
+            using (HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, _queryUri))
+            {
+                string accessToken = await GetAccessToken().ConfigureAwait(false);
```

Comment:
Nit: since this will most likely be used by our customers as a sample, the code should implement a Http DelegatingHandler to perform auth functions.